### PR TITLE
Add ability to force the size which an image node is rendered.

### PR DESCRIPTION
**Pull request by garrettmoon**
Putting this up to hopefully facilitate discussion.

### CODE REVIEW

**Comment by garrettmoon** [comment]
@appleguy would love your thoughts on this!

**Comment by maicki** [comment]
LGTM

**Comment by Adlai-Holler** [comment]
Would you be willing to add a unit test for this in ASImageNodeSnapshotTests that asserts that the node looks right (the snapshot) and that the dimensions of `contents` are as expected?

File: AsyncDisplayKit/Private/ASImageNode+CGExtras.m [hunk 1]
```diff
@@ -66,7 +67,10 @@ void ASCroppedImageBackingSizeAndDrawRectInBounds(CGSize sourceImageSize,
 
   // If fitting the desired aspect ratio to the image size actually results in a larger buffer, use the input values.
   // However, if there is a pixel savings (e.g. we would have to upscale the image), overwrite the function arguments.
-  if (forceUpscaling == NO && (scaledSizeForImage.width * scaledSizeForImage.height) < (destinationWidth * destinationHeight)) {
+  if (CGSizeEqualToSize(CGSizeZero, forcedSize) == NO) {
+    destinationWidth = forcedSize.width;
+    destinationHeight = forcedSize.height;
```
Comment (Adlai-Holler) [inline AsyncDisplayKit/Private/ASImageNode+CGExtras.m]: `round` this like below

**Comment by Adlai-Holler** [comment]
Great! This is a problem I had at Tripstr as well so it's good to see it in the bank!

**Pull request merged**